### DISCUSSION TITLE
Documentation of Map/Reduce index functions

### DISCUSSION THREAD

**Discussion by dbblackdiamond** [ravendb/ravendb#16913] [Q&A] [answered]
Hi,

I have been scouring the RavenDB documentation and haven't managed to find what I am looking for. I am looking for a document with the definition of all the functions that can be used within a Map/Reduce index. For instance, based on the documentation, the following functions can be used: `g.Sum`, `g.Min`, `g.Max`, and `g.Average`. What else is available? Where is the documentation explaining all the available functions?

Thanks a lot in advance,
Bertrand.

**Comment by ayende** [comment, marked as answer]
Basically, and Linq operation would be viable here. 

Full listing is here:

https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.ienumerable-1?view=net-7.0#extension-methods

Many of them don't actually make sense in map/reduce, by the way.
For example `Average` will not give you the values you want, in the context of map/reduce.
